Lib/argparse.py

```diff
@@ -1433,11 +1433,17 @@ def add_argument(self, *args, **kwargs):
                 kwargs['default'] = self.argument_default
 
         # create the action object, and add it to the parser
+        action_name = kwargs.get('action')
         action_class = self._pop_action_class(kwargs)
         if not callable(action_class):
             raise ValueError('unknown action "%s"' % (action_class,))
         action = action_class(**kwargs)
 
+        # raise an error if action for positional argument does not
+        # consume arguments
+        if not action.option_strings and action.nargs == 0:
+            raise ValueError(f'action {action_name!r} is not valid for positional arguments')
```

Comment:
I wonder if we could make this error more descriptive or actionable for the user. 

What do you think about adding something like  "Consider removing nargs=0 or converting the argument to optional if no input is required." at the end? Maybe something shorter?

Reply:
If `nargs=0` is explicitly specified, the user will get a different error: 'nargs for positionals must be != 0' (see `_get_positional_kwargs()`).

This exception is only raised when no `nargs` was specified, but the action sets nargs=0 in its constructor. The error message is in line with other error messages for invalid arguments of `add_argument()` (see below, and also `_get_positional_kwargs()` and `_get_optional_kwargs()`). It already mentions positional arguments, so mentioning optionals would be redundant.

The problem was that this limitation was not explicitly documented, so the user was not sure whether the issue is with their code or a bug in argparse. With explicit exceptions it will be more clear. Also, I think that it is worth to update the documentation (but this is a PR #125302).

Reply:
Makes sense - thanks for clarifying!